og.module

```diff
@@ -312,13 +312,48 @@ function og_invalidate_group_content_cache_tags(EntityInterface $entity) {
   // group listings to be cached effectively. Also invalidate the tags if the
   // group itself changes. The cache tag format is
   // 'og-group-content:{group entity type}:{group entity id}'.
-  if (Og::isGroup($entity->getEntityTypeId(), $entity->bundle()) || Og::isGroupContent($entity->getEntityTypeId(), $entity->bundle())) {
+  $is_group_content = Og::isGroupContent($entity->getEntityTypeId(), $entity->bundle());
+  $group_is_updated = Og::isGroup($entity->getEntityTypeId(), $entity->bundle()) && $entity->original;
+  if ($group_is_updated || $is_group_content) {
+    /** @var \Drupal\og\MembershipManagerInterface $membership_manager */
+    $membership_manager = \Drupal::service('og.membership_manager');
     $tags = [];
-    foreach (\Drupal::service('og.membership_manager')->getGroupIds($entity) as $entity_type => $ids) {
-      foreach ($ids as $id) {
-        $tags[] = "$entity_type:$id";
+
+    // If the entity is a group content and we came here as an effect of an
+    // update, check if any of the OG audience fields has been changed (AKA 'the
+    // group of the entity has changed') and add also the tags of the old group.
+    /** @var \Drupal\Core\Entity\FieldableEntityInterface $entity */
+    if ($is_group_content && $original = $entity->original) {
```

Comment:
The `&&` operator has higher precedence than the `=` operator but this still works (probably because the operation doesn't make sense otherwise). Let's move this out of the `if` statement to save others some confusion while reading this.